### PR TITLE
Add Kover Code Coverage

### LINKED ISSUE #2722: Setup code coverage CI check using Kover
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.

### What would you like to improve?

- Be able to report code coverage on the changes of each PR
- Being able to get a code coverage report for the project

### Because...

- We need to be motivated to write more tests.
- We need to know which codepaths are tested.

### Description

Setup Kover.
Add Kover as a job that runs after successful unit tests.

### Success Criteria

- Reports code coverage 
- Script to be able to run locally 
- Uploads code coverage report
- Runs as a job after Unit tests (success)
- Excludes the Ui Composables from the code coverage check
- It's free and open-source 
- We're legally okay to use the tool for free

### PR DESCRIPTION
**Pull request by SURAJ2600**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the "Contributing Rules".
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [No] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

Describe with a few bullets **what's new:**
- a Added [Kover](https://github.com/Kotlin/kotlinx-kover) for generating code coverage report
- b Added a new job in the existing workflow to upload  the generated code coverage reports in GitHub artifact 
- c The code coverage report will be run in CI only if the unit test workflow completed
- d using `./gradlew koverHtmlReport` will generate the report locally 

## Risk Factors

**What may go wrong if we merge your PR?**

- a None
- b
- c

**In what cases your code won't work?**

- a None
- b
- c

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/2722)**.

- Closes #2722

> Replace `2722` with your issue number (for example Closes #1234). If you've done that correctly, you'll see the issue title linked when previewing your PR description.

## Troubleshooting CI failures

If you see any of the PR checks failing (❌) go to [Actions](https://github.com/Ivy-Apps/ivy-wallet/actions) and find it there. Or simply click "Details" next to the failed check and explore the logs to see why it has failed.

### Detekt
[Detekt](https://detekt.dev/) is a static code analyzer for Kotlin that we use to enforce code readibility and good practices.

**To run Detekt locally:**
```
./gradlew detekt
```

If the Detekt errors are caused by a legacy code, you can suppress them using a basline.

**Detekt baseline** (not recommended)
```
./scripts/detektBaseline.sh
```

### Lint

We use the [standard Android Lint](https://developer.android.com/studio/write/lint) plus [Slack's compose-lints](https://slackhq.github.io/compose-lints/) as an addition to enforce proper Compose usage.

**To run Lint locally:**
```
./scripts/lint.sh
```

If the Lint errors are caused by a legacy code, you can suppress them using a basline.

**Lint baseline** (not recommended)
```
./scripts/lintBaseline.sh
```

### Unit tests

If this job is failing this means that your changes break an existing unit test. You must identify the failing tests and fix your code.

**To run the Unit tests locally:**
```
./gradlew testDebugUnitTest
```

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -186,3 +186,46 @@ dependencies {
...
+        defaults {
+            mergeWith("debug")
+            html {
+                onCheck = false
+                setReportDir(layout.buildDirectory.dir("artifacts/reports/kover/coverageResults"))
+            }
+        }
+    }
+
+    kover(projects.ivyBase)
```
Comment (ILIYANGERMANOV) [inline app/build.gradle.kts:201]: Can we automatically add all subprojects here?
Maybe using `projects.forEach` or something similar

File: build.gradle.kts [hunk 2]
```diff
@@ -6,6 +6,28 @@ plugins {
...
+    alias(libs.plugins.koverPlugin)
+}
+
+subprojects {
+    apply(plugin = "org.jetbrains.kotlinx.kover")
+    koverReport {
+        // filters for all report types of all build variants
+        filters {
+            excludes {
+                classes(
```
Comment (ILIYANGERMANOV) [inline build.gradle.kts:19]: Excellent work 💯

**Review by ILIYANGERMANOV — APPROVED** [review]
This is amazing!! Very well done 💯 🚀 
I'll merge it right away.

A few things that can be improved in a follow-up PR:
- Encapsulate all Kover build logic in `buildSrc/ivy.kover.gradle.kts` convention plugin
- Automatically add all modules to the report instead of `kover(..)`

**Comment by ILIYANGERMANOV** [comment]
Hey @SURAJ2600 do you have how hard would it be to enforce certain code coverage % for diffs in new PRs?

I want to enforce people to write unit tests

**Comment by SURAJ2600** [comment]
> This is amazing!! Very well done 💯 🚀 I'll merge it right away.
> 
> A few things that can be improved in a follow-up PR:
> 
> * Encapsulate all Kover build logic in `buildSrc/ivy.kover.gradle.kts` convention plugin
> * Automatically add all modules to the report instead of `kover(..)`

@ILIYANGERMANOV Thanks for the feedback, Sure I will take a look at it, thanks for merging my first PR. ❤️

**Comment by SURAJ2600** [comment]
> Hey @SURAJ2600 do you have how hard would it be to enforce certain code coverage % for diffs in new PRs?
> 
> I want to enforce people to write unit tests

@ILIYANGERMANOV I think we can make the CI pipeline fail if the code coverage is not up to some percentage.

**Pull request merged**